variables.tf

```diff
@@ -80,3 +80,28 @@ variable "write_as_files" {
   description = "Whether to write the policies as files to disk"
   default     = false
 }
+
+variable "enable_ecs_role_assumption" {
+  type        = bool
+  description = "Indicates whether the roles should allow assumption from ECS tasks."
+  default     = true
+}
+
+variable "enable_kubernetes_role_assumption" {
+  type        = bool
+  description = "Indicates whether the roles should allow assumption from Kubernetes service accounts."
+  default     = false
+}
```

Comment:
Perhaps having them as `locals` based on kubernetes config?

enable_ecs_role_assumption = var. kubernetes_config == null
enable_kubernetes_role_assumption = var. kubernetes_config != null

Reply:
The reason I didn't do that was because I wrote the trust relationships so that you could enable both if you wanted. But maybe there's no real point to doing it like that.

Reply:
hm, in what cases would you want both?

Reply:
I think it's unlikely you would, so I've refactored it. It just made sense in my head like that when I was initially writing it.